File: client/src/services/activity.ts
import { apiFetch } from './api';

export interface ActivityItem {
  id: string;
  kind: 'prank' | 'member_joined';
  at: string;
  actor_name?: string | null;
  target_name?: string | null;
  overlay_type?: string | null;
  status?: string | null;
  text?: string | null;
}

export async function listRoomActivity(
  roomId: string,
  limit = 40,
): Promise<ActivityItem[]> {
  const res = await apiFetch<{ items: ActivityItem[] }>(
    `/v1/rooms/${roomId}/activity?limit=${limit}`,
  );
  return res.items;
}

File: crates/screenraid-types/src/activity.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum ActivityKind {
    Prank,
    MemberJoined,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ActivityItem {
    pub id: Uuid,
    pub kind: ActivityKind,
    pub at: DateTime<Utc>,
    #[serde(default)]
    pub actor_name: Option<String>,
    #[serde(default)]
    pub target_name: Option<String>,
    #[serde(default)]
    pub overlay_type: Option<String>,
    #[serde(default)]
    pub status: Option<String>,
    #[serde(default)]
    pub text: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ActivityListResponse {
    pub items: Vec<ActivityItem>,
}
